Gist/Managers/MessageQueueManager.swift

```diff
@@ -2,61 +2,70 @@ import Foundation
 
 class MessageQueueManager {
     private var queueTimer: Timer!
-
+    
     func setup() {
         queueTimer = Timer.scheduledTimer(timeInterval: 10,
                                           target: self,
                                           selector: #selector(checkForMessages),
                                           userInfo: nil,
                                           repeats: true)
-        checkForMessages()
+        
+        // Since on app launch there's a short period where the applicationState is still set to "background"
+        // We wait 1 second for the app to become active before checking for messages.
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            self.checkForMessages()
+        }
```

Comment:
Out of curiosity, is there a way, we can get an instance of the `scene phase` that we can observe instead of a 1-second check? 

I know in swift-ui we can also observe notification center to know if the app is in foreground or background, maybe we can use that as a trigger to start/stop synching?

Reply:
It should be possible to observe changes, at this point we're only using this to make sure the app has transitioned into active in case Gist is initialized the first thing on start.

If we eventually go down this route we can use the notification to actually start and stop the queue listener as well.